src/abax-client.ts

```diff
@@ -322,11 +320,17 @@ export class AbaxClient {
 
   /** Recursively list all pages starting from the provided page number. Uses page size 1500 (maximum). */
   private async listNextPagesOfTrips(
-    input: { query: Omit<ListTripsInput['query'], 'page' | 'page_size'> },
+    input: {
+      dateFrom: Date;
+      dateTo: Date;
+      vehicleId?: string;
+    },
```

Comment:
I think in this case it might be correct to "be clever" and use an Omit type, since this method is a depagination of the original query, not some separate function that is detached from the input type. Any input fields added to the input type necessarily must be included here.